public/javascript/posts/BookAppointment.js

```diff
@@ -278,31 +278,29 @@ var app = angular.module('BookAppointmentApp');
     };
 
     $scope.getInitialData = function(){
-      $http.get('../api/calendar/calendar_details', {params:
-        {doclocation_id: $scope.locationId,
-         start: $scope.viewStartDate,
-         end: $scope.viewEndDate,
-         doctor_id: $scope.doctorId
-        }})
-        .success(function (response) {
-          var minTime = response.calendar.min.toString(),
-          maxTime = response.calendar.max.toString(), slot = null,
-          docMaxTime = null, docMinTime = null;
-          slot = response.calendar.slot_duration
-          $scope.uiConfig.calendar.slotDuration = slot;
-          slot = moment(slot, 'hh:mm:ss');
-          $rootScope.slot = (slot.hour() * 60) + slot.minutes();
-          docMinTime = minTime.slice(0, -2) + ":" + minTime.slice(-2);
-          docMaxTime = maxTime.slice(0, -2) + ":" + maxTime.slice(-2);
-          $scope.uiConfig.calendar.minTime = docMinTime;
-          $scope.uiConfig.calendar.maxTime = docMaxTime;
-          $rootScope.minTime = moment(docMinTime, 'hh:mm').format('hh:mm a');
-          $rootScope.maxTime = moment(docMaxTime, 'hh:mm').format('hh:mm a');
-          $scope.events.splice(0,$scope.events.length);
-          $scope.each(response.events, function(event){
-            $scope.events.push($scope.formatEvent(event));
+      if($scope.startDate !== $scope.viewStartDate || $scope.endDate !== $scope.viewEndDate || 
+        $scope.changedLocationId !== $scope.locationId || $scope.changedDoctorId !== $scope.doctorId){
+        $scope.startDate = $scope.viewStartDate;
+        $scope.endDate = $scope.viewEndDate;
+        $scope.changedLocationId = $scope.locationId;
+        $scope.changedDoctorId = $scope.doctorId;
+        $http.get('../api/calendar/calendar_details', {params:
+          {doclocation_id: $scope.locationId,
+           start: $scope.viewStartDate,
+           end: $scope.viewEndDate
+          }})
+          .success(function (response) {
+            var minTime = response.calendar.min.toString(),

```

Comment:
Code inside suceess function is changed , please check what is the latest code in master